src/components/TheCanvas.vue

```diff
@@ -5,52 +5,57 @@
     :config="stageSize"
     @mousedown="onMouseDownHandler"
     @mouseup="onMouseUpHandler"
-    @mousemove="throttleToolRender"
+    @mousemove="onMouseMoveHandler"
   >
     <v-layer ref="layer"/>
   </v-stage>
 </template>
 
 <script lang="ts">
-/* eslint-disable @typescript-eslint/no-this-alias */
-
 import { Prop } from 'vue-property-decorator'
 import Component from 'vue-class-component'
 import { Tool } from '@/tools/Tool'
-import throttle from 'lodash.throttle'
+import { VueKonvaLayer, VueKonvaStage, CanvasElementPayload, CanvasElement } from '@/types/Canvas'
 import Vue from 'vue'
 import { Namespaces } from '@/store'
 import { Action, Getter, namespace } from 'vuex-class'
 import { ToolGetters, ToolsAction } from '@/store/modules/tools'
+import { CanvasAction, CanvasGetters } from '@/store/modules/canvas'
 import Konva from 'konva'
 import { SocketActions, SocketGetters } from '@/store/modules/socket'
 
 const Tools = namespace(Namespaces.TOOLS)
 const Sockets = namespace(Namespaces.SOCKET)
 
-  @Component({
-    name: 'TheCanvas'
-  })
+@Component({
+  name: 'TheCanvas'
+})
 export default class TheCanvas extends Vue {
     @Prop() private id!: string
     @Tools.Action(ToolsAction.DISABLE) disable!: () => void
     @Action(`tools/${ToolsAction.ENABLE}`) enable!: () => void
-    @Action(`tools/${ToolsAction.ENABLE_TOOL}`) enableTool !: (name: string) => void
+    @Action(`tools/${ToolsAction.ENABLE_TOOL}`) enableTool!: (name: string) => void
+    @Getter(`tools/${ToolGetters.ENABLED_TOOL}`) enabledTool!: Tool
     @Getter(`tools/${ToolGetters.ENABLED}`) enabled!: boolean
     @Getter(`tools/${ToolGetters.TOOLS}`) tools!: Tool[]
-    @Getter(`tools/${ToolGetters.ACTIVE_TOOL}`) activeTool!: Tool
     @Sockets.Getter(SocketGetters.SOCKET) socket!: WebSocket
     @Sockets.Action(SocketActions.SEND_IF_OPEN) send!: (message: string) => void
+    @Action(`canvas/${CanvasAction.ADD_CANVAS_ELEMENT}`) addCanvasElement!: (canvasElementPayload: CanvasElementPayload) => void
+    @Getter(`canvas/${CanvasGetters.CANVAS_ELEMENTS}`) canvasElements!: () => void
 
     stageSize = {
       width: window.innerWidth,
       height: window.innerHeight
     }
+
+    canvasElementPayload = {
```

Comment:
You could define here Typescript types for cleaner readability.

Reply:
Yeah fair point :) though it isn't used in this file, it is only passed in, but I will get that changed.